Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,139 @@
+
+    private static final FilterPermanent anotherVampireFilter = new FilterControlledCreaturePermanent("another nontoken Vampire you control");
+    static {
+        anotherVampireFilter.add(AnotherPredicate.instance);
+        anotherVampireFilter.add(SubType.VAMPIRE.getPredicate());
+    }
+
+    public TimotharBaronOfBats(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{B}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.VAMPIRE);
+        this.addSubType(SubType.NOBLE);
+
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Ward—Discard a card.
+        this.addAbility(new WardAbility(new DiscardCardCost()));
+
+        // Whenever another nontoken Vampire you control dies, you may pay {1} and exile it.
+        // If you do, create a 1/1 black Bat creature token with flying.
+        // It gains “When this creature deals combat damage to a player,
+        // sacrifice it and return the exiled card to the battlefield tapped.”
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new DoIfCostPaid(new TimotharBaronOfBatsCreateBatEffect(), new ManaCostsImpl<>("{1}"))
+                        .setText("you may pay {1} and exile it. " +
+                                "If you do, create a 1/1 black Bat creature token with flying. " +
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.EXILED, source, game);
+
+        BatToken bat = new BatToken();
+        bat.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
```

Comment:
Not sure if this is correct.  Usually you would use `game.addEffect` on a `GainAbilityTargetEffect` for this type of thing but I guess since it's a token and it's a permanent gain effect it may work out to do the same thing in practice.

Reply:
I don't follow.

Reply:
Like you did with Stefan's second ability "gains indestructible until end of turn" except using Duration.Custom to make it a permanent gain.

Not sure if it matters or not in this case though.

Reply:
Oh I see. Does anyone here know if it makes a difference?

And if I should change it. Do I add it to the bat before calling `CreateTokenEffect(bat, 1)`?